apps/system/update/binary_update.c

```diff
@@ -0,0 +1,248 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/***************************************************************************
+ * Included Files
+ ***************************************************************************/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <errno.h>
+#include <sched.h>
+#include <time.h>
+#include <fcntl.h>
+#include <string.h>
+#include <pthread.h>
+#include <sys/time.h>
+#include <mqueue.h>
+#include <tinyara/binary_manager.h>
+
+#include "binary_update.h"
+
+static int send_requset(binmgr_request_t *request_msg)
+{
+	int ret;
+	mqd_t binmgr_mq;
+	binmgr_mq = mq_open(BINMGR_REQUEST_MQ, O_WRONLY, 0666, 0);
+	if (binmgr_mq == (mqd_t)ERROR) {
+		mq_close(binmgr_mq);
+		dbg("mq_open failed!\n");
```

Comment:
You should not use `dbg`. You should make this module specific debug messages or use printf.